.github/workflows/go-main.yml

```diff
@@ -46,6 +44,38 @@ jobs:
         with:
           go-version: 1.16
 
+      - name: Get branch name for push
```

Comment:
we don't need to get the branch name now, we know it's 'main' so just set it if necessary